src/ecmascript/spidermonkey/localstorage-db.c

```diff
@@ -189,15 +171,13 @@ db_query_by_key(char *db_name, char *key)
 		return("");
 	}
 	sqlite3_busy_timeout(db, 2000);
-	sqlbuffer=stracpy("SELECT * FROM storage WHERE key like ?;");
-	rc=sqlite3_prepare_v2(db, sqlbuffer, strlen(sqlbuffer) + 1, &stmt, NULL);
-	rc=sqlite3_bind_text(stmt, 1, key, strlen(key) + 1, SQLITE_STATIC);
+	rc=sqlite3_prepare_v2(db, "SELECT * FROM storage WHERE key = ?;", -1, &stmt, NULL);
+	rc=sqlite3_bind_text(stmt, 1, key, strlen(key), SQLITE_STATIC);
+	result=stracpy("");
 	while (sqlite3_step(stmt) == SQLITE_ROW) {
```

Comment:
Instead of while - if. You can also add LIMIT 1 to SELECT.
If you change it, I'm going  to merge it.

Reply:
Here You are. Thank You very much. ;]